akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowMapAsyncSpec.scala

```diff
@@ -295,6 +295,27 @@ class FlowMapAsyncSpec extends StreamSpec {
         3.seconds) should ===("happy!")
     }
 
+    "complete without requiring further demand" in assertAllStagesStopped {
```

Comment:
Isn't the fix for MapAsync still missing?

Reply:
It's optimized to never use mapAsync for parallelism 1 so I didn't bother (not sure the bug is there as well either)

Reply:
Ah, I missed that.

Reply:
For good measure I now double checked that the `MapAsync` graphstage is internal, so shouldn't be use by anyone in for example GraphDSL either.

Reply:
It starts failing with `parallelism = 2`, we need tests for those as well.

Reply:
Ah, thanks, I didn't think of that. I'll look into it.

Reply:
Added coverage for both and fixed